### PR TITLE
new(ci): add a test-drivers-ppc64le workflow.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area CI

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Adds a ppc64le drivers_test CI.
It leverages https://github.com/appleboy/ssh-action since ppc64le is currently unsupported by github actions self-hosted runner: https://github.com/actions/runner/issues/2365 (https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#architectures).

There are of course some security implications: we are giving ssh access to our ppc64le node to a gha, that, while being of course open source, can be victim of attacks.
While i can see that this is quite a big issue, we:
* only use this node to test the build of drivers
* we do not generate any artifact on the node

Given the above, i think it might be a good solution until gha self-hosted runners support ppc64le natively.

cc @LucaGuerra 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

~~This is still wip; still have to add proper secrets to the libs repo.~~

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/drivers_ci.yml [hunk 1]
```diff
@@ -173,17 +173,40 @@ jobs:
...
+    steps:
+      - name: Build and test drivers on ppc64le node via ssh
+        if: needs.paths-filter.outputs.driver_needs_rebuild
+        uses: appleboy/ssh-action@v1.0.3
+        with:
+          host: ${{ secrets.PPC64LE_HOST }}
+          username: ${{ secrets.PPC64LE_USERNAME }}
+          key: ${{ secrets.PPC64LE_KEY }}
+          port: ${{ secrets.PPC64LE_PORT }}
+          envs: GITHUB_REF_NAME,GITHUB_REPOSITORY,GITHUB_SERVER_URL
```
Comment (FedeDP) [inline .github/workflows/drivers_ci.yml:190]: Shared envs.

File: .github/workflows/drivers_ci.yml [hunk 2]
```diff
@@ -173,17 +173,40 @@ jobs:
...
+          script: |
+            sudo dnf install -y bpftool ca-certificates cmake make automake gcc gcc-c++ kernel-devel clang git pkg-config autoconf automake libbpf-devel
```
Comment (FedeDP) [inline .github/workflows/drivers_ci.yml:192]: The node runs fedora OS.

File: .github/workflows/drivers_ci.yml [hunk 3]
```diff
@@ -173,17 +173,40 @@ jobs:
...
+            temp_fld=$(mktemp -d)
```
Comment (FedeDP) [inline .github/workflows/drivers_ci.yml:193]: Each invokation runs inside a temp folder. This  way we support parallel runs.

File: .github/workflows/drivers_ci.yml [hunk 4]
```diff
@@ -173,17 +173,40 @@ jobs:
...
+            pushd $temp_fld
+            git clone -b $GITHUB_REF_NAME $GITHUB_SERVER_URL/$GITHUB_REPOSITORY.git libs
```
Comment (FedeDP) [inline .github/workflows/drivers_ci.yml:195]: Clone the current branch under the temp_fld/libs.

**Comment by FedeDP** [comment]
I will add the secrets and then trigger the drivers ci with a fake driver code change.

**Comment by FedeDP** [comment]
7 tests failing for kmod:
```
out: [  FAILED  ] 7 tests, listed below:
out: [  FAILED  ] SyscallExit.clone3X_child
out: [  FAILED  ] SyscallExit.clone3X_child_clone_parent_flag
out: [  FAILED  ] SyscallExit.clone3X_child_new_namespace_from_child
out: [  FAILED  ] SyscallExit.execveX_failure
out: [  FAILED  ] SyscallExit.execveX_success
out: [  FAILED  ] SyscallExit.execveatX_failure
out: [  FAILED  ] SyscallExit.execveatX_correct_exit
```
and 3 for old bpf:
```
out: [  FAILED  ] 3 tests, listed below:
out: [  FAILED  ] SyscallExit.clone3X_child
out: [  FAILED  ] SyscallExit.clone3X_child_clone_parent_flag
out: [  FAILED  ] SyscallExit.clone3X_child_new_namespace_from_child
```

File: .github/workflows/drivers_ci.yml [hunk 5]
```diff
@@ -173,17 +173,44 @@ jobs:
...
+            mkdir -p build
+            cd build && cmake -DUSE_BUNDLED_DEPS=ON -DENABLE_DRIVERS_TESTS=ON -DBUILD_LIBSCAP_MODERN_BPF=ON -DMODERN_BPF_DEBUG_MODE=ON -DBUILD_BPF=True -DBUILD_LIBSCAP_GVISOR=OFF ../
+            make drivers_test driver bpf -j6
+            sudo ./test/drivers/drivers_test -m
+            rc_modern=$?
+            sudo ./test/drivers/drivers_test -b
+            rc_bpf=$?
+            sudo ./test/drivers/drivers_test -k
+            rc_kmod=$?
+            exit $(($rc_modern + $rc_bpf +$rc_kmod))
```
Comment (FedeDP) [inline .github/workflows/drivers_ci.yml:208]: Exit with RC!=0 if any of {modern, bpf, kmod } tests fail.

File: driver/main.c
Comment (FedeDP) [inline driver/main.c:1]: Changes will be reverted before removing `wip`.

**Comment by FedeDP** [comment]
```
[ RUN      ] SyscallExit.clone3X_child
/home/fedora/libs/test/drivers/event_class/event_class.cpp:972: Failure
Failed
There is no event '335' in the buffers.

[  FAILED  ] SyscallExit.clone3X_child (61 ms)
```

**Comment by incertum** [comment]
Is this ready @FedeDP? LGTM. It runs smoothly?

**Comment by FedeDP** [comment]
Still wip because i need to find the time to fix 
>  Drivers CI Build / test-drivers-ppc64le 😁 (system_deps,custom node) (pull_request) Failing after 10m 

:) 
As aforementioned, we have 3 tests failing.

File: driver/bpf/fillers.h [hunk 6]
```diff
@@ -2600,9 +2600,12 @@ FILLER(proc_startupdate_3, true)
 			flags = bpf_syscall_get_argument(data, 0);
 			if (bpf_probe_read_user(&cl_args, sizeof(struct clone_args), (void *)flags)) 
 			{
-				return PPM_FAILURE_INVALID_USER_MEMORY;
+				flags = 0;
```
Comment (FedeDP) [inline driver/bpf/fillers.h:2603]: Same behavior as modern bpf: https://github.com/falcosecurity/libs/blob/master/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/clone3.bpf.c#L191

File: driver/ppm_fillers.c [hunk 7]
```diff
@@ -1207,9 +1207,12 @@ int f_proc_startupdate(struct event_filler_arguments *args)
 			res = ppm_copy_from_user(&cl_args, (void *)val, sizeof(struct clone_args));
 			if (unlikely(res != 0))
 			{
-				return PPM_FAILURE_INVALID_USER_MEMORY;
+				val = 0;
```
Comment (FedeDP) [inline driver/ppm_fillers.c:1210]: Same behavior as modern bpf: https://github.com/falcosecurity/libs/blob/master/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/clone3.bpf.c#L191

**Comment by FedeDP** [comment]
OK bpf and modern bpf are fixed.
We now have 4 failing tests on kmod:
```
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] SyscallExit.execveX_failure
[  FAILED  ] SyscallExit.execveX_success
[  FAILED  ] SyscallExit.execveatX_failure
[  FAILED  ] SyscallExit.execveatX_correct_exit
```
they are probably "caused" by changes from #1800 ; perhaps there is some weird page fault somewhere, or a small behavior difference from other drivers?

**Comment by FedeDP** [comment]
Yes :point_up: the issue is caused by `PAGE_SIZE` being `65536` on ppc64le, that is  being used as `STR_STORAGE_SIZE` (hard size limit on kmod) that is different from `MAX_PROC_ARG_ENV` (4096) used by bpf drivers.
That's why only kmod is failing.
Modifying the hard limit on kmod is pretty risky; i will fix the tests.

**Comment by FedeDP** [comment]
Removed wip.
@incertum this is now ready since ppc64 CI is :green_circle: ! 
/unhold

**Review by incertum — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
Rebased on master to fix conflict.

**Review by incertum — APPROVED** [review]
/approve

**Review by incertum — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Pull request merged**